src/submission/submission-service.ts

```diff
@@ -641,6 +646,72 @@ export namespace operations {
     );
   };
 
+  export async function getAllCommittedClinicalData(programId: string) {
+    if (!programId) throw new Error('Missing programId!');
+
+    // this query takes about 6sec for 20,000 donors with sample registration only running locally
+    const donors = await getDonors(programId); // get in batch??
+
+    // collect all records
+    const recordsMap: any = {};
+    donors.forEach(d => {
+      Object.values(ClinicalEntitySchemaNames).forEach(entity => {
+        let clincialInfoRecords;
+        if (entity === ClinicalEntitySchemaNames.REGISTRATION) {
+          clincialInfoRecords = generateSampleRegistrationRecordsFromDonor(d);
+        } else {
+          clincialInfoRecords = getClinicalEntitiesFromDonorBySchemaName(d, entity);
+        }
+        recordsMap[entity] = _.concat(recordsMap[entity] || [], clincialInfoRecords);
+      });
+    });
+
+    // map into object ready for api processing
+    const schemasWithFields = schemaManager.instance().getSchemasWithFields();
```

Comment:
do we actually need the schema? shouldn't we export whatever is in the clinicalInfo field ?

Reply:
that can cause some inconsistency in fields if the latest is different from the donor doc in some fields